File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/models/PackageSummary.java
package io.github.thunkier.thunkmetrc.wrapper.models;import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;public class PackageSummary {
    @JsonProperty("ArchivedDate")
    public String archivedDate;
    @JsonProperty("ContainsDecontaminatedProduct")
    public Boolean containsDecontaminatedProduct;
    @JsonProperty("ContainsRemediatedProduct")
    public Boolean containsRemediatedProduct;
    @JsonProperty("DecontaminationDate")
    public String decontaminationDate;
    @JsonProperty("ExpirationDate")
    public String expirationDate;
    @JsonProperty("ExternalId")
    public Integer externalId;
    @JsonProperty("FinishedDate")
    public String finishedDate;
    @JsonProperty("Id")
    public Integer id;
    @JsonProperty("InitialLabTestingState")
    public String initialLabTestingState;
    @JsonProperty("IsDonation")
    public Boolean isDonation;
    @JsonProperty("IsDonationPersistent")
    public Boolean isDonationPersistent;
    @JsonProperty("IsFinished")
    public Boolean isFinished;
    @JsonProperty("IsOnHold")
    public Boolean isOnHold;
    @JsonProperty("IsOnHoldCombined")
    public Boolean isOnHoldCombined;
    @JsonProperty("IsOnInvestigation")
    public Boolean isOnInvestigation;
    @JsonProperty("IsOnInvestigationHold")
    public Boolean isOnInvestigationHold;
    @JsonProperty("IsOnInvestigationRecall")
    public Boolean isOnInvestigationRecall;
    @JsonProperty("IsOnRecall")
    public Boolean isOnRecall;
    @JsonProperty("IsOnRecallCombined")
    public Boolean isOnRecallCombined;
    @JsonProperty("IsOnRetailerDelivery")
    public Boolean isOnRetailerDelivery;
    @JsonProperty("IsProcessValidationTestingSample")
    public Boolean isProcessValidationTestingSample;
    @JsonProperty("IsProductionBatch")
    public Boolean isProductionBatch;
    @JsonProperty("IsTestingSample")
    public Boolean isTestingSample;
    @JsonProperty("IsTradeSample")
    public Boolean isTradeSample;
    @JsonProperty("IsTradeSamplePersistent")
    public Boolean isTradeSamplePersistent;
    @JsonProperty("Item")
    public PackagesPackageSummaryItem item;
    @JsonProperty("ItemFromFacilityLicenseNumber")
    public String itemFromFacilityLicenseNumber;
    @JsonProperty("ItemFromFacilityName")
    public String itemFromFacilityName;
    @JsonProperty("LabTestResultExpirationDateTime")
    public String labTestResultExpirationDateTime;
    @JsonProperty("LabTestStage")
    public String labTestStage;
    @JsonProperty("LabTestStageId")
    public Integer labTestStageId;
    @JsonProperty("LabTestingPerformedDate")
    public String labTestingPerformedDate;
    @JsonProperty("LabTestingRecordedDate")
    public String labTestingRecordedDate;
    @JsonProperty("LabTestingState")
    public String labTestingState;
    @JsonProperty("LabTestingStateDate")
    public String labTestingStateDate;
    @JsonProperty("Label")
    public String label;
    @JsonProperty("LabelsLastGeneratedDateTime")
    public String labelsLastGeneratedDateTime;
    @JsonProperty("LastModified")
    public String lastModified;
    @JsonProperty("LocationId")
    public Integer locationId;
    @JsonProperty("LocationName")
    public String locationName;
    @JsonProperty("LocationTypeName")
    public String locationTypeName;
    @JsonProperty("Note")
    public String note;
    @JsonProperty("OriginalPackageQuantity")
    public Double originalPackageQuantity;
    @JsonProperty("PackageForProductDestruction")
    public String packageForProductDestruction;
    @JsonProperty("PackageType")
    public String packageType;
    @JsonProperty("PackagedDate")
    public String packagedDate;
    @JsonProperty("PatientLicenseNumber")
    public String patientLicenseNumber;
    @JsonProperty("ProductLabel")
    public PackagesPackageSummaryProductLabel productLabel;
    @JsonProperty("ProductRequiresDecontamination")
    public Boolean productRequiresDecontamination;
    @JsonProperty("ProductRequiresRemediation")
    public Boolean productRequiresRemediation;
    @JsonProperty("ProductionBatchNumber")
    public String productionBatchNumber;
    @JsonProperty("Quantity")
    public Double quantity;
    @JsonProperty("ReceivedDateTime")
    public String receivedDateTime;
    @JsonProperty("ReceivedFromFacilityLicenseNumber")
    public String receivedFromFacilityLicenseNumber;
    @JsonProperty("ReceivedFromFacilityName")
    public String receivedFromFacilityName;
    @JsonProperty("ReceivedFromManifestNumber")
    public String receivedFromManifestNumber;
    @JsonProperty("RemediationDate")
    public String remediationDate;
    @JsonProperty("SellByDate")
    public String sellByDate;
    @JsonProperty("SourceHarvestCount")
    public Integer sourceHarvestCount;
    @JsonProperty("SourceHarvestNames")
    public String sourceHarvestNames;
    @JsonProperty("SourcePackageCount")
    public Integer sourcePackageCount;
    @JsonProperty("SourcePackageIsDonation")
    public Boolean sourcePackageIsDonation;
    @JsonProperty("SourcePackageIsTradeSample")
    public Boolean sourcePackageIsTradeSample;
    @JsonProperty("SourcePackageLabels")
    public String sourcePackageLabels;
    @JsonProperty("SourceProcessingJobCount")
    public Integer sourceProcessingJobCount;
    @JsonProperty("SourceProductionBatchNumbers")
    public String sourceProductionBatchNumbers;
    @JsonProperty("SublocationId")
    public Integer sublocationId;
    @JsonProperty("SublocationName")
    public String sublocationName;
    @JsonProperty("UnitOfMeasureAbbreviation")
    public String unitOfMeasureAbbreviation;
    @JsonProperty("UnitOfMeasureName")
    public String unitOfMeasureName;
    @JsonProperty("UseByDate")
    public String useByDate;
    public static class PackagesPackageSummaryItem {
    @JsonProperty("AdministrationMethod")
    public String administrationMethod;
    @JsonProperty("Allergens")
    public String allergens;
    @JsonProperty("ApprovalStatus")
    public Integer approvalStatus;
    @JsonProperty("ApprovalStatusDateTime")
    public String approvalStatusDateTime;
    @JsonProperty("DefaultLabTestingState")
    public Integer defaultLabTestingState;
    @JsonProperty("Description")
    public String description;
    @JsonProperty("GlobalProductName")
    public String globalProductName;
    @JsonProperty("GlobalProductNumber")
    public String globalProductNumber;
    @JsonProperty("HasExpirationDate")
    public Boolean hasExpirationDate;
    @JsonProperty("HasSellByDate")
    public Boolean hasSellByDate;
    @JsonProperty("HasUseByDate")
    public Boolean hasUseByDate;
    @JsonProperty("Id")
    public Integer id;
    @JsonProperty("IsExpirationDateRequired")
    public Boolean isExpirationDateRequired;
    @JsonProperty("IsSellByDateRequired")
    public Boolean isSellByDateRequired;
    @JsonProperty("IsUseByDateRequired")
    public Boolean isUseByDateRequired;
    @JsonProperty("IsUsed")
    public Boolean isUsed;
    @JsonProperty("ItemBrandId")
    public Integer itemBrandId;
    @JsonProperty("ItemBrandName")
    public String itemBrandName;
    @JsonProperty("LabTestBatchNames")
    public List<Object> labTestBatchNames;
    @JsonProperty("LabelImages")
    public List<Object> labelImages;
    @JsonProperty("LabelPhotoDescription")
    public String labelPhotoDescription;
    @JsonProperty("Name")
    public String name;
    @JsonProperty("NumberOfDoses")
    public String numberOfDoses;
    @JsonProperty("PackagingImages")
    public List<Object> packagingImages;
    @JsonProperty("PackagingPhotoDescription")
    public String packagingPhotoDescription;
    @JsonProperty("ProcessingJobCategoryName")
    public String processingJobCategoryName;
    @JsonProperty("ProcessingJobTypeName")
    public String processingJobTypeName;
    @JsonProperty("ProductBrandName")
    public String productBrandName;
    @JsonProperty("ProductCategoryName")
    public String productCategoryName;
    @JsonProperty("ProductCategoryType")
    public Integer productCategoryType;
    @JsonProperty("ProductImages")
    public List<Object> productImages;
    @JsonProperty("ProductPDFDocuments")
    public List<Object> productPDFDocuments;
    @JsonProperty("ProductPhotoDescription")
    public String productPhotoDescription;
    @JsonProperty("PublicIngredients")
    public String publicIngredients;
    @JsonProperty("QuantityType")
    public Integer quantityType;
    @JsonProperty("ServingSize")
    public String servingSize;
    @JsonProperty("StrainId")
    public Integer strainId;
    @JsonProperty("StrainName")
    public String strainName;
    @JsonProperty("UnitCbdAContent")
    public Double unitCbdAContent;
    @JsonProperty("UnitCbdAContentDose")
    public Double unitCbdAContentDose;
    @JsonProperty("UnitCbdAContentDoseUnitOfMeasureName")
    public String unitCbdAContentDoseUnitOfMeasureName;
    @JsonProperty("UnitCbdAContentUnitOfMeasureName")
    public String unitCbdAContentUnitOfMeasureName;
    @JsonProperty("UnitCbdAPercent")
    public Double unitCbdAPercent;
    @JsonProperty("UnitCbdContent")
    public Double unitCbdContent;
    @JsonProperty("UnitCbdContentDose")
    public Double unitCbdContentDose;
    @JsonProperty("UnitCbdContentDoseUnitOfMeasureName")
    public String unitCbdContentDoseUnitOfMeasureName;
    @JsonProperty("UnitCbdContentUnitOfMeasureName")
    public String unitCbdContentUnitOfMeasureName;
    @JsonProperty("UnitCbdPercent")
    public Double unitCbdPercent;
    @JsonProperty("UnitOfMeasureName")
    public String unitOfMeasureName;
    @JsonProperty("UnitQuantity")
    public Double unitQuantity;
    @JsonProperty("UnitQuantityUnitOfMeasureName")
    public String unitQuantityUnitOfMeasureName;
    @JsonProperty("UnitThcAContent")
    public Double unitThcAContent;
    @JsonProperty("UnitThcAContentDose")
    public Double unitThcAContentDose;
    @JsonProperty("UnitThcAContentDoseUnitOfMeasureId")
    public Integer unitThcAContentDoseUnitOfMeasureId;
    @JsonProperty("UnitThcAContentDoseUnitOfMeasureName")
    public String unitThcAContentDoseUnitOfMeasureName;
    @JsonProperty("UnitThcAContentUnitOfMeasureName")
    public String unitThcAContentUnitOfMeasureName;
    @JsonProperty("UnitThcAPercent")
    public Double unitThcAPercent;
    @JsonProperty("UnitThcContent")
    public Double unitThcContent;
    @JsonProperty("UnitThcContentDose")
    public Double unitThcContentDose;
    @JsonProperty("UnitThcContentDoseUnitOfMeasureId")
    public Integer unitThcContentDoseUnitOfMeasureId;
    @JsonProperty("UnitThcContentDoseUnitOfMeasureName")
    public String unitThcContentDoseUnitOfMeasureName;
    @JsonProperty("UnitThcContentUnitOfMeasureName")
    public String unitThcContentUnitOfMeasureName;
    @JsonProperty("UnitThcPercent")
    public Double unitThcPercent;
    @JsonProperty("UnitVolume")
    public Double unitVolume;
    @JsonProperty("UnitVolumeUnitOfMeasureName")
    public String unitVolumeUnitOfMeasureName;
    @JsonProperty("UnitWeight")
    public Double unitWeight;
    @JsonProperty("UnitWeightUnitOfMeasureName")
    public String unitWeightUnitOfMeasureName;
}
    public static class PackagesPackageSummaryProductLabel {
    @JsonProperty("IsActive")
    public Boolean isActive;
    @JsonProperty("IsChildFromParentWithLabel")
    public Boolean isChildFromParentWithLabel;
    @JsonProperty("IsDiscontinued")
    public Boolean isDiscontinued;
    @JsonProperty("LastLabelGenerationDate")
    public String lastLabelGenerationDate;
    @JsonProperty("PackageId")
    public Integer packageId;
    @JsonProperty("QrCount")
    public Integer qrCount;
    @JsonProperty("TagId")
    public Integer tagId;
}
}

File: sdks/thunkmetrc-rs/wrapper/src/models/package_summary.rs
use serde::{Deserialize, Serialize};
use super::*;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PackageSummary {
    #[serde(rename = "ArchivedDate", skip_serializing_if = "Option::is_none")]
    pub archived_date: Option<String>,
    #[serde(rename = "ContainsDecontaminatedProduct", skip_serializing_if = "Option::is_none")]
    pub contains_decontaminated_product: Option<bool>,
    #[serde(rename = "ContainsRemediatedProduct", skip_serializing_if = "Option::is_none")]
    pub contains_remediated_product: Option<bool>,
    #[serde(rename = "DecontaminationDate", skip_serializing_if = "Option::is_none")]
    pub decontamination_date: Option<String>,
    #[serde(rename = "ExpirationDate", skip_serializing_if = "Option::is_none")]
    pub expiration_date: Option<String>,
    #[serde(rename = "ExternalId", skip_serializing_if = "Option::is_none")]
    pub external_id: Option<i64>,
    #[serde(rename = "FinishedDate", skip_serializing_if = "Option::is_none")]
    pub finished_date: Option<String>,
    #[serde(rename = "Id", skip_serializing_if = "Option::is_none")]
    pub id: Option<i64>,
    #[serde(rename = "InitialLabTestingState", skip_serializing_if = "Option::is_none")]
    pub initial_lab_testing_state: Option<String>,
    #[serde(rename = "IsDonation", skip_serializing_if = "Option::is_none")]
    pub is_donation: Option<bool>,
    #[serde(rename = "IsDonationPersistent", skip_serializing_if = "Option::is_none")]
    pub is_donation_persistent: Option<bool>,
    #[serde(rename = "IsFinished", skip_serializing_if = "Option::is_none")]
    pub is_finished: Option<bool>,
    #[serde(rename = "IsOnHold", skip_serializing_if = "Option::is_none")]
    pub is_on_hold: Option<bool>,
    #[serde(rename = "IsOnHoldCombined", skip_serializing_if = "Option::is_none")]
    pub is_on_hold_combined: Option<bool>,
    #[serde(rename = "IsOnInvestigation", skip_serializing_if = "Option::is_none")]
    pub is_on_investigation: Option<bool>,
    #[serde(rename = "IsOnInvestigationHold", skip_serializing_if = "Option::is_none")]
    pub is_on_investigation_hold: Option<bool>,
    #[serde(rename = "IsOnInvestigationRecall", skip_serializing_if = "Option::is_none")]
    pub is_on_investigation_recall: Option<bool>,
    #[serde(rename = "IsOnRecall", skip_serializing_if = "Option::is_none")]
    pub is_on_recall: Option<bool>,
    #[serde(rename = "IsOnRecallCombined", skip_serializing_if = "Option::is_none")]
    pub is_on_recall_combined: Option<bool>,
    #[serde(rename = "IsOnRetailerDelivery", skip_serializing_if = "Option::is_none")]
    pub is_on_retailer_delivery: Option<bool>,
    #[serde(rename = "IsProcessValidationTestingSample", skip_serializing_if = "Option::is_none")]
    pub is_process_validation_testing_sample: Option<bool>,
    #[serde(rename = "IsProductionBatch", skip_serializing_if = "Option::is_none")]
    pub is_production_batch: Option<bool>,
    #[serde(rename = "IsTestingSample", skip_serializing_if = "Option::is_none")]
    pub is_testing_sample: Option<bool>,
    #[serde(rename = "IsTradeSample", skip_serializing_if = "Option::is_none")]
    pub is_trade_sample: Option<bool>,
    #[serde(rename = "IsTradeSamplePersistent", skip_serializing_if = "Option::is_none")]
    pub is_trade_sample_persistent: Option<bool>,
    #[serde(rename = "Item", skip_serializing_if = "Option::is_none")]
    pub item: Option<PackagesPackageSummaryItem>,
    #[serde(rename = "ItemFromFacilityLicenseNumber", skip_serializing_if = "Option::is_none")]
    pub item_from_facility_license_number: Option<String>,
    #[serde(rename = "ItemFromFacilityName", skip_serializing_if = "Option::is_none")]
    pub item_from_facility_name: Option<String>,
    #[serde(rename = "LabTestResultExpirationDateTime", skip_serializing_if = "Option::is_none")]
    pub lab_test_result_expiration_date_time: Option<String>,
    #[serde(rename = "LabTestStage", skip_serializing_if = "Option::is_none")]
    pub lab_test_stage: Option<String>,
    #[serde(rename = "LabTestStageId", skip_serializing_if = "Option::is_none")]
    pub lab_test_stage_id: Option<i64>,
    #[serde(rename = "LabTestingPerformedDate", skip_serializing_if = "Option::is_none")]
    pub lab_testing_performed_date: Option<String>,
    #[serde(rename = "LabTestingRecordedDate", skip_serializing_if = "Option::is_none")]
    pub lab_testing_recorded_date: Option<String>,
    #[serde(rename = "LabTestingState", skip_serializing_if = "Option::is_none")]
    pub lab_testing_state: Option<String>,
    #[serde(rename = "LabTestingStateDate", skip_serializing_if = "Option::is_none")]
    pub lab_testing_state_date: Option<String>,
    #[serde(rename = "Label", skip_serializing_if = "Option::is_none")]
    pub label: Option<String>,
    #[serde(rename = "LabelsLastGeneratedDateTime", skip_serializing_if = "Option::is_none")]
    pub labels_last_generated_date_time: Option<String>,
    #[serde(rename = "LastModified", skip_serializing_if = "Option::is_none")]
    pub last_modified: Option<String>,
    #[serde(rename = "LocationId", skip_serializing_if = "Option::is_none")]
    pub location_id: Option<i64>,
    #[serde(rename = "LocationName", skip_serializing_if = "Option::is_none")]
    pub location_name: Option<String>,
    #[serde(rename = "LocationTypeName", skip_serializing_if = "Option::is_none")]
    pub location_type_name: Option<String>,
    #[serde(rename = "Note", skip_serializing_if = "Option::is_none")]
    pub note: Option<String>,
    #[serde(rename = "OriginalPackageQuantity", skip_serializing_if = "Option::is_none")]
    pub original_package_quantity: Option<f64>,
    #[serde(rename = "PackageForProductDestruction", skip_serializing_if = "Option::is_none")]
    pub package_for_product_destruction: Option<String>,
    #[serde(rename = "PackageType", skip_serializing_if = "Option::is_none")]
    pub package_type: Option<String>,
    #[serde(rename = "PackagedDate", skip_serializing_if = "Option::is_none")]
    pub packaged_date: Option<String>,
    #[serde(rename = "PatientLicenseNumber", skip_serializing_if = "Option::is_none")]
    pub patient_license_number: Option<String>,
    #[serde(rename = "ProductLabel", skip_serializing_if = "Option::is_none")]
    pub product_label: Option<PackagesPackageSummaryProductLabel>,
    #[serde(rename = "ProductRequiresDecontamination", skip_serializing_if = "Option::is_none")]
    pub product_requires_decontamination: Option<bool>,
    #[serde(rename = "ProductRequiresRemediation", skip_serializing_if = "Option::is_none")]
    pub product_requires_remediation: Option<bool>,
    #[serde(rename = "ProductionBatchNumber", skip_serializing_if = "Option::is_none")]
    pub production_batch_number: Option<String>,
    #[serde(rename = "Quantity", skip_serializing_if = "Option::is_none")]
    pub quantity: Option<f64>,
    #[serde(rename = "ReceivedDateTime", skip_serializing_if = "Option::is_none")]
    pub received_date_time: Option<String>,
    #[serde(rename = "ReceivedFromFacilityLicenseNumber", skip_serializing_if = "Option::is_none")]
    pub received_from_facility_license_number: Option<String>,
    #[serde(rename = "ReceivedFromFacilityName", skip_serializing_if = "Option::is_none")]
    pub received_from_facility_name: Option<String>,
    #[serde(rename = "ReceivedFromManifestNumber", skip_serializing_if = "Option::is_none")]
    pub received_from_manifest_number: Option<String>,
    #[serde(rename = "RemediationDate", skip_serializing_if = "Option::is_none")]
    pub remediation_date: Option<String>,
    #[serde(rename = "SellByDate", skip_serializing_if = "Option::is_none")]
    pub sell_by_date: Option<String>,
    #[serde(rename = "SourceHarvestCount", skip_serializing_if = "Option::is_none")]
    pub source_harvest_count: Option<i64>,
    #[serde(rename = "SourceHarvestNames", skip_serializing_if = "Option::is_none")]
    pub source_harvest_names: Option<String>,
    #[serde(rename = "SourcePackageCount", skip_serializing_if = "Option::is_none")]
    pub source_package_count: Option<i64>,
    #[serde(rename = "SourcePackageIsDonation", skip_serializing_if = "Option::is_none")]
    pub source_package_is_donation: Option<bool>,
    #[serde(rename = "SourcePackageIsTradeSample", skip_serializing_if = "Option::is_none")]
    pub source_package_is_trade_sample: Option<bool>,
    #[serde(rename = "SourcePackageLabels", skip_serializing_if = "Option::is_none")]
    pub source_package_labels: Option<String>,
    #[serde(rename = "SourceProcessingJobCount", skip_serializing_if = "Option::is_none")]
    pub source_processing_job_count: Option<i64>,
    #[serde(rename = "SourceProductionBatchNumbers", skip_serializing_if = "Option::is_none")]
    pub source_production_batch_numbers: Option<String>,
    #[serde(rename = "SublocationId", skip_serializing_if = "Option::is_none")]
    pub sublocation_id: Option<i64>,
    #[serde(rename = "SublocationName", skip_serializing_if = "Option::is_none")]
    pub sublocation_name: Option<String>,
    #[serde(rename = "UnitOfMeasureAbbreviation", skip_serializing_if = "Option::is_none")]
    pub unit_of_measure_abbreviation: Option<String>,
    #[serde(rename = "UnitOfMeasureName", skip_serializing_if = "Option::is_none")]
    pub unit_of_measure_name: Option<String>,
    #[serde(rename = "UseByDate", skip_serializing_if = "Option::is_none")]
    pub use_by_date: Option<String>,
}
